packages/utils/src/compat/css.ts

```diff
@@ -0,0 +1,172 @@
+                    return getUnsupportedBrowsers(entry, tokenPrefix, browsers);
+                }
+            }
+        }
+
+        if (matches.regex_token) {
+            const regexToken = matches.regex_token && new RegExp(matches.regex_token);
+
+            for (const [tokenPrefix, tokenValue] of tokens) {
+                if (regexToken && regexToken.exec(tokenValue)) {
+                    return getUnsupportedBrowsers(entry, tokenPrefix, browsers);
+                }
+            }
+        }
+    }
+
+    return null;
+};
+
+/**
+ * Determine if the provided CSS value is supported, first by looking for an
+ * exact match for the full value, falling back to search for a partial match.
+ */
+const getValueUnsupported = (context: Identifier, value: string, browsers: string[]): UnsupportedBrowsers => {
+    const [data, prefix] = getFeatureData(context, value);
+
+    if (data) {
+        return getUnsupportedBrowsers(data, prefix, browsers);
+    }
+
+    return getPartialValueUnsupported(context, value, browsers);
+};
+
+/**
+ * Determine if the provided CSS declaration consisting of a property
+ * and optionally a value is supported (e.g. `border-radius` or `display: grid`).
+ */
+export const getDeclarationUnsupported = (feature: DeclarationQuery, browsers: string[]): UnsupportedBrowsers => {
+    const key = `css-declaration:${feature.property}|${feature.value || ''}`;
+
+    return getCachedValue(key, browsers, () => {
+        const [data, prefix] = getFeatureData(mdn.css.properties, feature.property);
+
+        if (feature.value) {
+            return getValueUnsupported(data, feature.value, browsers);
+        }
+
+        return getUnsupportedBrowsers(data, prefix, browsers);
+    });
+};
+
+/**
+ * Determine if the provided CSS at-rule is supported (e.g. `keyframes`).
+ */
+export const getRuleUnsupported = (feature: RuleQuery, browsers: string[]): UnsupportedBrowsers => {
+    return getCachedValue(`css-rule:${feature.rule}`, browsers, () => {
+        const [data, prefix] = getFeatureData(mdn.css['at-rules'], feature.rule);
+
+        return getUnsupportedBrowsers(data, prefix, browsers);
+    });
+};
+
+const getPseudoSelectorUnsupported = (value: string, browsers: string[]): UnsupportedBrowsers => {
+    const name = value.replace(/^::?/, ''); // Strip leading `:` or `::`.
+
+    return getCachedValue(`css-pseudo-selector:${name}`, browsers, () => {
+        const [data, prefix] = getFeatureData(mdn.css.selectors, name);
+
+        return getUnsupportedBrowsers(data, prefix, browsers);
+    });
+};
+
+/**
+ * Determine if the provided CSS selector is supported (e.g. `:valid`).
+ *
+ * Can be passed any selector, but currently only validates pseudo-selectors
+ * as most other selector types already have broad support and certain other
+ * special cases (e.g. newer attribute and combinator selectors) need special
+ * handling to map to the MDN data (which hasn't been done yet).
```

Comment:
Is there an issue in MDN for this? If so maybe we should link it? Maybe create another issue in our side to know we have to do this and link to the MDN issue?